test/metadata_tests.cpp

```diff
@@ -84,6 +85,30 @@ BOOST_AUTO_TEST_CASE( head_bucket ) {
     free_client(client);
 }
 
+BOOST_AUTO_TEST_CASE( head_folder ) {
+    printf("-----Testing head_folder-------\n");
+    ds3_error* error;
+    ds3_client* client = get_client();
+    const char* bucket_name = "head_folder_test";
+    ds3_request* request = ds3_init_put_bucket(bucket_name);
+    error = ds3_put_bucket(client, request);
+    ds3_free_request(request);
+    handle_error(error);
+
+    const char* test_folder = "test_folder/";
+    request = ds3_init_put_object_for_job(bucket_name, test_folder, 0, 0, NULL);
+    error   = ds3_put_object(client, request, NULL, NULL);
+    ds3_free_request(request);
+    handle_error(error);
+
+    request = ds3_init_head_bucket(bucket_name);
+    error = ds3_head_bucket(client, request);
```

Comment:
One method might be to check that either there's a slash at the end AND that the string only contains two slashes? Bucket names aren't allowed to contain slashes, so if there's more two, you know an object name was passed cause that's where the extra slash has to come from.